keymanapp/keyman
# fix(windows): check IM window will be in a visible location

The IM display call back is used to show the IM window. With this change deadcode is removed and a check is made to ensure the IM window has completley visible coordinates for the monitor that has focus. If not and only if not it will be position close to where the caret in the application that has focus.


## USER Testing

## TEST_IMX_VISIBLE

To do this test, you need two monitors. 

1. Install the Keyman for Windows Build with this PR
2. Install the Simplified Chinese cs_pinyin keyboard
3. Open Notepad
4. Type <kbd>h</kbd><kbd>a</kbd><kbd>n</kbd><kbd>z</kbd><kbd>i</kbd> 
5. Ensure the IM pop-up appears and is visible
6. In the Windows settings for monitors if it is not already set primary monitor to be on the right of the second monitor. In Windows 10 this is the setting
![image](https://github.com/user-attachments/assets/ea1fa63c-c655-4b20-ab59-0c8c476387d5)

8. Go to Notepad and again type 
9. Insure the IMX is still visible
